lib/astutils.cpp

```diff
@@ -1023,6 +1023,73 @@ static bool isZeroBoundCond(const Token * const cond)
     return false;
 }
 
+static bool isForLoopCondition(const Token * const tok)
+{
+    if (!tok || !tok->astParent())
+        return false;
+    const Token *const parent = tok->astParent();
+    return parent->str() == ";" && parent->astOperand1() == tok &&
+           parent->astParent() && parent->astParent()->str() == ";" &&
```

Comment:
Please use Token::simpleMatch when it makes sense.

Instead of `parent->astParent() && parent->astParent()->str() == ";"` I would prefer `Token::simpleMatch(parent->astParent(), ";")`.